v2: panic unmarshaling inline table to string

**Describe the bug**
```
panic: subslice address (0) is before data address (824636779408)
goroutine 1 [running]:
github.com/pelletier/go-toml/v2/internal/danger.SubsliceOffset({0xc0002eab90, 0x8, 0x8}, {0x0, 0x0, 0x0})
        C:/Users/Nathan/go/pkg/mod/github.com/pelletier/go-toml/v2@v2.0.6/internal/danger/danger.go:16 +0x273
github.com/pelletier/go-toml/v2.wrapDecodeError({0xc0002eab90, 0x8, 0x8}, 0xc0002cd0c0)
        C:/Users/Nathan/go/pkg/mod/github.com/pelletier/go-toml/v2@v2.0.6/errors.go:91 +0x5c
github.com/pelletier/go-toml/v2.(*decoder).FromParser(0xc000207f00, {0xea01a0?, 0xc00000a4b8?})
        C:/Users/Nathan/go/pkg/mod/github.com/pelletier/go-toml/v2@v2.0.6/unmarshaler.go:216 +0x24c
github.com/pelletier/go-toml/v2.Unmarshal({0xc0002eab90, 0x8, 0x8}, {0xea01a0, 0xc00000a4b8})
        C:/Users/Nathan/go/pkg/mod/github.com/pelletier/go-toml/v2@v2.0.6/unmarshaler.go:28 +0x11a
main.main()
```

**To Reproduce**
```go
data := make(map[string]string)
_ = toml.Unmarshal([]byte("foo = {}"), &data)
```

**Expected behavior**
A meaningful error.

**Versions**
 - go-toml: v2.0.6
 - go: 1.19
 - operating system: Windows

**Additional context**
Similar to #581 and #588.

--- a/unmarshaler.go
+++ b/unmarshaler.go
@@ -746,7 +746,7 @@ func (d *decoder) unmarshalInlineTable(itable *unstable.Node, v reflect.Value) e
 		}
 		return d.unmarshalInlineTable(itable, elem)
 	default:
-		return unstable.NewParserError(itable.Data, "cannot store inline table in Go type %s", v.Kind())
+		return unstable.NewParserError(d.p.Raw(itable.Raw), "cannot store inline table in Go type %s", v.Kind())
 	}
 
 	it := itable.Children()

--- a/unmarshaler_test.go
+++ b/unmarshaler_test.go
@@ -2466,6 +2466,12 @@ func TestIssue807(t *testing.T) {
 	require.Equal(t, "foo", m.Name)
 }
 
+func TestIssue850(t *testing.T) {
+	data := make(map[string]string)
+	err := toml.Unmarshal([]byte("foo = {}"), &data)
+	require.Error(t, err)
+}
+
 func TestUnmarshalDecodeErrors(t *testing.T) {
 	examples := []struct {
 		desc string

--- a/unstable/parser.go
+++ b/unstable/parser.go
@@ -402,6 +402,7 @@ func (p *Parser) parseInlineTable(b []byte) (reference, []byte, error) {
 	// inline-table-keyvals = keyval [ inline-table-sep inline-table-keyvals ]
 	parent := p.builder.Push(Node{
 		Kind: InlineTable,
+		Raw:  p.Range(b[:1]),
 	})
 
 	first := true

